Code/Framework/AzCore/AzCore/Math/Sfmt.h

```diff
@@ -106,10 +106,10 @@ namespace AZ
         void    PeriodCertification();
 
         SfmtInternal::w128_t        m_sfmt[SfmtInternal::N];
-        AZStd::atomic_int           m_index;   ///  Index into the pre-generated tables
+        size_t                      m_index;   ///  Index into the pre-generated tables
```

Comment:
This index isn't initialized at all.
I recommend initializing the `m_index` and `m_psfmt32` and `m_psfmt64` variables to 0 and nullptr